### PR TITLE
Replace old sp model with new ODB model (partial implementation)

### PR DESCRIPTION
**Pull request by jluhrs**

### CODE REVIEW

File: modules/engine/src/main/scala/observe/engine/Sequence.scala [hunk 1]
```diff
@@ -10,25 +10,33 @@ import observe.engine.Action.ActionState
 import observe.engine.Result.RetVal
 import observe.model.SequenceState
 import lucuma.core.model.Observation
-import lucuma.core.model.sequence.Step.{Id => StepId}
+import lucuma.core.model.sequence.Atom as OcsAtom
```
Comment (rpiaggio) [inline modules/engine/src/main/scala/observe/engine/Sequence.scala:13]: Is the alias needed?

File: modules/engine/src/main/scala/observe/engine/Sequence.scala [hunk 2]
```diff
@@ -10,25 +10,33 @@ import observe.engine.Action.ActionState
...
 
 /**
  * A list of `Step`s grouped by target and instrument.
  */
-final case class Sequence[F[_]](
-  id:    Observation.Id,
-  steps: List[Step[F]]
+final case class Sequence[F[_]] private (
+  id:     Observation.Id,
+  atomId: Option[OcsAtom.Id],
```
Comment (rpiaggio) [inline modules/engine/src/main/scala/observe/engine/Sequence.scala:21]: The `Sequence` is a flattened list of `Steps`? Why is the `Atom` optional?

**Review by rpiaggio — APPROVED** [review]
LGTM. TBH, I haven't looked in depth at all the changes.

File: modules/engine/src/main/scala/observe/engine/Sequence.scala (hunk 2 above)
Comment (jluhrs) [reply modules/engine/src/main/scala/observe/engine/Sequence.scala]: To allow for the special case of an empty Sequence. For example, one that has run to completion.

**Pull request merged**